### PR TITLE
Shell script tidy up.

### PR DESCRIPTION
**Pull request by CrossR**
I've tidied up the existing `setup.sh` script as well as implementing a few of the TODOs.
I also added an equivalent script for PowerShell, since (IMO) asking for WSL just to run a single script seems overkill.

Assuming it looks good, I can add it to the docs as well, should just be a case of running

```
iex ((New-Object System.Net.WebClient).DownloadString('https://raw.githubusercontent.com/CrossR/zmk/CrossR/ShellScriptTidyUp/docs/static/setup.ps1'))
```

The main question I had, which I expect someone has already thought about, is how to deal with the growing list of boards/shields and multiple sources of truth? Its already a pain when there is the source code and the bash script, and now there is a third source. Having a single location and deriving these two scripts from it seems best, though probably a pain in bash/powershell. That question doesn't need to be answered now though.

Potentially easier in something like Python, but I also like that the setup so far only really needs git so adding a dependency on Python could be a pain for Windows.

### CODE REVIEW

**Comment by petejohanson** [comment]
> The main question I had, which I expect someone has already thought about, is how to deal with the growing list of boards/shields and multiple sources of truth? Its already a pain when there is the source code and the bash script, and now there is a third source. Having a single location and deriving these two scripts from it seems best, though probably a pain in bash/powershell. That question doesn't need to be answered now though.
> 
> Potentially easier in something like Python, but I also like that the setup so far only really needs git so adding a dependency on Python could be a pain for Windows.

@CrossR I had originally envisioned, once this got big enough, or we added a PS script, would be change the `setup.sh` to actually be `setup.sh.in` and have it be templated, so that we could process it at "build time" for the docs site, and inject the current set of shields, their settings, etc. there.

If you're up for trying to take a stab at that, that would be great. If not, we can create an issue to capture that plan.

Either way, I'm not sure we should *block* getting the PS script in on that work. I will try to do a detailed review on this PR as time permits.

Thanks!

**Review by petejohanson — COMMENTED** [review]
Two minor nitpicks. Can you give me a screenshot/log of using the PS script, since I can't easily test myself?

Thanks for working on this!

**Comment by CrossR** [comment]
Comments should be fixed, and I've stuck a quick screen recording up here : https://drive.google.com/file/d/1LO6qa4gElXLTmvc-pRoput84V-pyciRY/view?usp=sharing and the repo it made is here : 
https://github.com/CrossR/temp_zmk_config.

Could be worth someone else on Windows giving it a quick run as well?

File: docs/static/setup.ps1 [hunk 1]
```diff
@@ -0,0 +1,189 @@
+$ErrorActionPreference = "Stop"
```
Comment (petejohanson) [inline docs/static/setup.ps1:1]: Can you please add a copyright and license header here?

File: docs/static/setup.sh [hunk 2]
```diff
@@ -1,14 +1,25 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (petejohanson) [inline docs/static/setup.sh:1]: While you are at it, can you please add a header here too after the magic start line?

**Review by petejohanson — COMMENTED** [review]
A couple minor header needs, then this is good to go. Thanks.

File: docs/static/setup.ps1 [hunk 3]
```diff
@@ -0,0 +1,193 @@
+# Copyright (c) 2020 Ryan Cross
```
Comment (petejohanson) [inline docs/static/setup.ps1:1]: Sorry I wasn't clear before, can we stick with the single author file approach please, e.g. 


```suggestion
# Copyright (c) 2020 The ZMK Contributors
```

File: docs/static/setup.sh [hunk 4]
```diff
@@ -1,14 +1,29 @@
...
+
+# Copyright (c) 2020 Pete Johanson, Ryan Cross
```
Comment (petejohanson) [inline docs/static/setup.sh:3]: Ditto.

**Review by petejohanson — CHANGES_REQUESTED** [review]
Sorry, one minor tweak, then this is good.

**Review by petejohanson — APPROVED** [review]
Thanks!

**Pull request merged**